app/controllers/sessions_controller.rb

```diff
@@ -10,22 +10,24 @@ def can_authenticate?(user)
 
   def create
     @user = User.find_by(email: params.dig(:session, :email)&.downcase)
-    if can_authenticate?(@user)
-      forwarding_url = session[:forwarding_url]
-      reset_session
-      log_in @user
-      params[:session][:remember_me] == '1' ? remember(@user) : forget(@user)
-      session[:session_token] = @user.session_token
-      redirect_to forwarding_url || @user
-    else
-      flash.now[:danger] = 'Invalid email/password combination'
-      render 'new'
-    end
-  end
 
-  def current_user
-    if session[:user_id]
-      User.find_by(id: session[:user_id])
+    if can_authenticate?(@user)
+      if @user.activated?
+          forwarding_url = session[:forwarding_url]
+          reset_session
+          log_in @user
+          params[:session][:remember_me] == '1' ? remember(@user) : forget(@user)
+          session[:session_token] = @user.session_token
+          redirect_to forwarding_url || @user
+        else
+          message = "Account not activated. "
```

Comment:
Please have this as one message, this is from Michael Hartl, but it's not necessary.